### DISCUSSION TITLE
Performance form

### DISCUSSION THREAD

**Discussion by SergeyWebDeveloper** [react-hook-form/react-hook-form#5373] [Q&A]
Hello, @bluebill1049. I faced the following problem at my job. I have reproduced an example sandbox situation - https://codesandbox.io/s/runtime-snowflake-ktzur?file=/src/App.js 
When you click on the add fields button, friezes are visible (this is especially noticeable in my work). Could you suggest how best to deal with this situation and improve form optimization when adding fields with `useFieldArray`?

Example performance in codesandbox form when adding `N` fields.
![image](https://user-images.githubusercontent.com/12486958/119897315-e4140580-bf48-11eb-80f7-6b1d89cb5818.png)

**Comment by bluebill1049** [comment]
This initial render can't be avoided, because you are rendering out all of those inputs. how many inputs are you try to render on the screen?

**Reply by bluebill1049** [reply]
another tip is try to use native input instead of custom input, it may improve the performance for the rendering of those inputs.

**Reply by SergeyWebDeveloper** [reply]
There can be any number of inputs since the form is dynamic.
